pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -175,6 +175,7 @@
 @Setter(AccessLevel.PROTECTED)
 public class BrokerService implements Closeable, ZooKeeperCacheListener<Policies> {
     private static final Logger log = LoggerFactory.getLogger(BrokerService.class);
+    private static final int CHANNEL_CLOSE_TIMEOUT_SECONDS = 10;
```

Comment:
Can you make it configurable?

Reply:
+1 for @sijie's proposal
in production the process should exit as soon as possible, when the process exists network resources are released

probably we can leave a little timeout by default (like 2 seconds) and set it to 10s in tests (if it is really needed)

Reply:
@sijie Thanks for the review. I've made it configurable in the most recent commit pushed to this PR.

Reply:
@eolivelli Thanks for the review. I've made the default to be 2 seconds. That should be sufficient for tests as well.

Reply:
I'm adding a short explanation here. The solution changed after the comment was made. The proposal by Matteo in https://github.com/apache/pulsar/pull/9308#discussion_r565390915 was to use the existing `brokerShutdownTimeoutMs` setting for the port closing. This resulted in the changes that are in the PR currently. I'd suggest reviewing the final changes.